lib/healthcheck/government_data_check.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Healthcheck
+  class GovernmentDataCheck
+    attr_reader :government_repo
+    def initialize
+      @government_repo = BulkData::GovernmentRepository.new
+    end
+
+    def name
+      :government_data_check
+    end
+
+    def status
+      return GovukHealthcheck::CRITICAL unless government_repo.cache_populated?
+      return GovukHealthcheck::WARNING if government_repo.cache_age > 6.hours
+
+      GovukHealthcheck::OK
+    end
+
+    def message
+      "No government data availible" if status == GovukHealthcheck::CRITICAL
+      warning_details_content if status == GovukHealthcheck:: WARNING
```

Comment:
This isn't going to work correctly when the status is critical as it isn't the last statement is the second conditional so this will return nil.

You'll need a return statement on the first line or to refactor this a bit. Probably best unit testing it though so this is covered.

Reply:
Absolutely right on this, I didn't realise that's how this worked :/ I've added in unit tests to make sure this gets caught.